testing/TestHarness/TestHarness.UITest/Ext/Navigation/PageNavigation/Given_PageNavigation.cs

```diff
@@ -168,6 +168,22 @@ public void When_PageNavigationRegisteredRoot()
 
 	}
 
+	[Test]
+	public void When_PageNavigationDataContextDidntChange()
+	{
+		InitTestSection(TestSections.Navigation_PageNavigation);
+
+		App.WaitThenTap("ShowOnePageButton");
+
+		App.WaitThenTap("OnePageGetUrlFromBrowser");
+		App.WaitElement("OnePageTxtDataContext");
+		var dataContextStatus = App.GetText("OnePageTxtDataContext");
+
+		// If DataContext is ever changed to anything other than the expected
+		// the text will be "DataContext is not correct"
+		Assert.AreEqual("DataContext is ok", dataContextStatus);
```

Comment:
You can use the `WaitForText` extension method instead of getting the text and asserting it separately https://github.com/unoplatform/uno.extensions/blob/ce25d2b2269955c48a374d29ed17bc94f0a0e011/testing/TestHarness/TestHarness.UITest/Extensions/QueryExtensions.cs#L16